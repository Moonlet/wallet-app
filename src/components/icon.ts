import { createIconSet } from 'react-native-vector-icons';
import * as glyphMap from '../../resources/fonts/icons.json';

Object.keys(glyphMap).map(key => {
    (glyphMap as any)[key] = parseInt(String((glyphMap as any)[key]).replace('\\f', '0xf'), 16);
});

export const Icon = createIconSet(glyphMap, 'icons', 'icons.ttf');
export default Icon;
