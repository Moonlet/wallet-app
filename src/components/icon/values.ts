export enum IconValues {
    // please try to add keys in alphabetic order
    ADD = 'add',
    ADD_CIRCLE = 'add-circle',
    ALARM_BELL = 'alarm-bell',
    ARCHIVE_LOCKER = 'archive-locker',
    ARROW_BUTTON_DOWN = 'arrow-button-down',
    ARROW_DOWN = 'arrow-down',
    ARROW_LEFT = 'arrow-left',
    ARROW_RIGHT_DASH = 'arrow-right-dash',
    ARROW_RIGHT = 'arrow-right',
    BIN = 'bin',
    BLE = 'ble',
    BUG = 'bug',
    BUTTON_REFRESH_ONE = 'button-refresh-one',
    CHECK_BOX_THICKED = 'check-box-thicked',
    CHECK_BOX = 'check-box',
    CHECK = 'check',
    CHEVRON_DOWN = 'chevron-down',
    CHEVRON_DOWN_BIGGER = 'chevron-down-2',
    CHEVRON_RIGHT = 'chevron-right',
    CHEVRON_UP = 'chevron-up',
    CLAIM_REWARD = 'claim-reward',
    CLOSE = 'close',
    COPY = 'copy',
    DASHBOARD = 'dashboard',
    EXPAND = 'expand',
    EYE = 'eye',
    FACE_ID = 'face-id',
    FAILED = 'failed',
    FLASH_OFF = 'flash-off',
    FLASH = 'flash',
    INBOUND = 'inbound',
    KEY_LOCK = 'key-lock',
    KEY = 'key',
    KEYBOARD_DELETE = 'keyboard-delete',
    KEYBOARD_SHIFT = 'keyboard-shift',
    LAPTOP = 'laptop',
    LEDGER_LOOGO = 'ledger-logo',
    MONEY_WALLET = 'money-wallet',
    MONITOR = 'monitor',
    MINUS = 'minus',
    NAVIGATION_MENU_HORIZONTAL = 'navigation-menu-horizontal',
    NAVIGATION_MENU_VERTICAL = 'navigation-menu-vertical',
    NAVIGATION_MENU = 'navigation-menu',
    NOTES_LIST = 'notes-list',
    OUTBOUND = 'outbound',
    PENCIL = 'pencil',
    PENDING = 'pending',
    PLUS = 'plus',
    QR_CODE_SCAN = 'qr-code-scan',
    REINVEST = 'reinvest',
    REVOTE = 'revote',
    SATURN_ICON = 'saturn-icon',
    SEARCH = 'search',
    SETTINGS = 'settings',
    SINGLE_MAN_HIERACHY = 'single-man-hierachy',
    TOUCH_ID = 'touch-id',
    UNLOCK = 'unlock',
    UNVOTE = 'unvote',
    USB = 'usb',
    VIEW_OFF = 'view-off',
    VIEW = 'view',
    VOTE = 'vote',
    WARNING = 'warning',
    WITHDRAW = 'withdraw'
    // please try to add keys in alphabetic order
}
