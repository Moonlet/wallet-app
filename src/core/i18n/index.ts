export * from './translation/translate-component';
export * from './translation/translate';
