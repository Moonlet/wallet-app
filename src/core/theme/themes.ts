export enum Theme {
    dark = 'dark',
    light = 'light'
}
