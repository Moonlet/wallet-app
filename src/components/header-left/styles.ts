import { StyleSheet } from 'react-native';
import { ITheme } from '../../core/theme/itheme';
import { BASE_DIMENSION, ICON_CONTAINER_SIZE } from '../../styles/dimensions';
import { normalize } from '../../library';

export default (theme: ITheme) =>
    StyleSheet.create({
        button: {
            flexDirection: 'row',
            alignItems: 'center'
        },
        iconContainer: {
            width: ICON_CONTAINER_SIZE,
            height: ICON_CONTAINER_SIZE,
            alignItems: 'center',
            justifyContent: 'center',
            marginLeft: BASE_DIMENSION
        },
        icon: {
            color: theme.colors.accent
        },
        text: {
            lineHeight: normalize(22),
            color: theme.colors.text
        }
    });
