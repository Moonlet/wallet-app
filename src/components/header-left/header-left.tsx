import React from 'react';
import { View, TouchableOpacity } from 'react-native';
import { Text, normalize } from '../../library';

import stylesProvider from './styles';
import { withTheme } from '../../core/theme/with-theme';
import { Icon } from '../icon';
import { ICON_SIZE } from '../../styles/dimensions';

export interface IProps {
    onPress?: any;
    icon?: string;
    text?: string;
    styles: ReturnType<typeof stylesProvider>;
}

export const HeaderLeftComponent = (props: IProps) => (
    <TouchableOpacity onPress={props.onPress} style={props.styles.button}>
        {props.icon && (
            <View style={props.styles.iconContainer}>
                <Icon name={props.icon} size={normalize(ICON_SIZE)} style={props.styles.icon} />
            </View>
        )}
        {props.text && <Text style={props.styles.text}>{props.text}</Text>}
    </TouchableOpacity>
);

export const HeaderLeft = withTheme(stylesProvider)(HeaderLeftComponent);
