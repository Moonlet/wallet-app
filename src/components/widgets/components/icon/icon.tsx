import React from 'react';
import { TouchableOpacity } from 'react-native';
import stylesProvider from './styles';
import { smartConnect } from '../../../../core/utils/smart-connect';
import { withTheme, IThemeProps } from '../../../../core/theme/with-theme';
import { IScreenModule, IIconData, ISmartScreenActions } from '../../types';
import { formatStyles } from '../../utils';
import Icon from '../../../icon/icon';
import { normalize } from '../../../../styles/dimensions';

interface IExternalProps {
    module: IScreenModule;
    actions: ISmartScreenActions;
    options?: {
        screenKey?: string;
        flowId?: string;
    };
}

const IconModuleComponent = (
    props: IThemeProps<ReturnType<typeof stylesProvider>> & IExternalProps
) => {
    const { module, styles } = props;
    const data = module.data as IIconData;

    const customStyle = formatStyles(module?.style);

    const moduleJSX = (
        <Icon
            name={data.icon}
            size={(customStyle?.size as number) || normalize(18)}
            style={[styles.icon, customStyle]}
        />
    );

    if (module?.cta) {
        return (
            <TouchableOpacity
                onPress={() => props.actions.handleCta(module.cta, props?.options)}
                activeOpacity={0.9}
            >
                {moduleJSX}
            </TouchableOpacity>
        );
    } else {
        return moduleJSX;
    }
};

export const IconModule = smartConnect<IExternalProps>(IconModuleComponent, [
    withTheme(stylesProvider)
]);
