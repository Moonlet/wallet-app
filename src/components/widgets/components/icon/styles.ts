import { StyleSheet } from 'react-native';
import { ITheme } from '../../../../core/theme/itheme';
import { BASE_DIMENSION } from '../../../../styles/dimensions';

export default (theme: ITheme) =>
    StyleSheet.create({
        icon: {
            alignSelf: 'center',
            color: theme.colors.textTertiary,
            padding: BASE_DIMENSION
        }
    });
