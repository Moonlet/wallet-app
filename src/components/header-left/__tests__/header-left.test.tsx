import React from 'react';
import { HeaderLeftComponent, IProps } from '../header-left';
import { darkTheme } from '../../../styles/themes/dark-theme';
import styleProvider from '../styles';

import { shallow } from 'enzyme';
import { loadTranslations } from '../../../core/i18n';
import { IconValues } from '../../icon/values';

const props: IProps = {
    styles: styleProvider(darkTheme),
    icon: IconValues.SETTINGS,
    text: 'text',
    onPress: jest.fn()
};

describe('header go back button', () => {
    beforeAll(async () => loadTranslations('en'));

    it('renders correctly', () => {
        const wrapper = shallow(<HeaderLeftComponent {...props} />);
        expect(wrapper.debug()).toMatchSnapshot();
    });
});
