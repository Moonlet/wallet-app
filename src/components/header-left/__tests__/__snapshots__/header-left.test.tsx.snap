// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header go back button renders correctly 1`] = `
"<TouchableOpacity onPress={[Function: mockConstructor]} style={{...}} activeOpacity={0.2}>
  <View style={{...}}>
    <Icon name=\\"settings\\" size={44} style={{...}} allowFontScaling={false} />
  </View>
  <withTheme(TextComponent) style={{...}}>
    text
  </withTheme(TextComponent)>
</TouchableOpacity>"
`;
